keras/utils/data_utils.py

```diff
@@ -654,7 +654,7 @@ def get(self):
                     yield inputs
             else:
                 all_finished = all([not thread.is_alive() for thread in self._threads])
-                if all_finished:
+                if all_finished and self.queue.empty():
```

Comment:
By checking for the second time if the queue is empty we can fix the race condition problem. If all the threads are inactive and the queue is empty, no further updates will happen on the queue. Thus it is safe to raise a StopIteration() exception. Nevertheless if all the threads are inactive and the last thread managed to update the queue, we will sleep and loop again until we exhaust the items of the queue.

Reply:
empty is not thread_safe. Please use self.queue.join()

Reply:
I think I check that all threads are not alive before that. I should be covered right?